### DISCUSSION TITLE
sequelize.connectionManager.releaseConnection() returns since update

### DISCUSSION THREAD

**Discussion by rconjaerts** [sequelize/sequelize#16156] [Help & Questions] [answered]
Yesterday, I finally made the step to update sequelize from v6.19 to v6.32. We've been postponing this because it took a lot of tedious work due to the `replacement`, `bind` issues that were fixed after v6.19.

Since I've updated this, I've been getting errors when I'm trying to release a connection.

```
sequelize.connectionManager
    .releaseConnection(conn)
    .then(() => null)
    .catch(err => {
        if (err && onError) onError(err);
        return null;
    });
```

Cannot read properties of undefined (reading 'then').

Is this new normal behaviour that this doesn't return a `Promise`? I wasn't able to find anything on this in the documentation.

**Comment by ephys** [comment, marked as answer]
Releasing a connection only marks it as available to the pool, and is a fully synchronous operation (just flags it as available for use by the pool).

It was accidentally marked as async for a long time

`destroyConnection` is still asynchronous, and kills the connection.

---

If you are trying to access connection objects manually, know that Sequelize 7 has a new api called `sequelize.withConnection` that gives you a connection for the duration of the callback: https://github.com/sequelize/sequelize/blob/64a1b82be246f4f6b39012fedebb1ce903128c7d/packages/core/src/sequelize-typescript.ts#L506
